### PR TITLE
fix(masthead): add padding top to masthead-composite & cloud-masthead-container

### PR DESCRIPTION
**Pull request by proeung**
### Related Ticket(s)

https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/6270

### Description

- This PR adds apply the padding-top value of `48px` to the masthead components (Global & Cloud) instead of the DotcomShell `<dds-dotcom-shell>` component.
- This allows adopters who are not using the `<dds-dotcom-shell>` component to render the masthead to not have to define the padding-top value within their application in order to fix the overlapping issue.

### Changelog

**New**

- The `padding-top: 48px;` value is being applied within the masthead composite updated function because CSS properties that are set in the host component are not being rendered.

**Testing**

- There should be no visual diffs for this change.

### CODE REVIEW

**Review by annawen1 — COMMENTED** [review]
@proeung thanks for putting this fix in! Noticing in the `DotcomShell` story that the padding doesn't seem to appearing

Screen Shot 2022-01-19 at 9 12 25 AM

File: packages/web-components/src/components/masthead/masthead-composite.ts [hunk 1]
```diff
@@ -770,6 +770,7 @@ class DDSMastheadComposite extends LitElement {
 
     // This is a temp fix until we figure out why we can't set styles to the :host(dds-masthead-container) in stylesheets
     this.style.zIndex = '900';
+    this.style.paddingTop = '48px';
```
[the remark this answers is not in the document]
Comment (annawen1) [reply packages/web-components/src/components/masthead/masthead-composite.ts:773]: I haven't been able to figure out this issue either, I've tried a couple of things but none have worked so far 😕

[the remark this answers is not in the document]
Comment (proeung) [reply packages/web-components/src/components/masthead/masthead-composite.ts:773]: Yeah, this is definitely an odd one. Regardless, I think the temporary fix should still work fine. Please review/test this PR and let me know if there's anything that I'm missing.

**Review by RichKummer — APPROVED** [review]
Thanks @proeung! LGTM!

**Review by annawen1 — APPROVED** [review]
LGTM!

**Pull request merged**